docs/madr/decisions/049-meshexternalservice-api.md

```diff
@@ -0,0 +1,518 @@
+        * **caCert**: defines a certificate of CA.
+          * one of `inline`, `inlineString` or `secret`.
+        * **clientCert**: defines a certificate of a client.
+          * one of `inline`, `inlineString` or `secret`.
+        * **clientKey**: defines a client private key.
+          * one of `inline`, `inlineString` or `secret`.
+* **status**: status of an object managed by Kuma control-plane
+  * **vip**: section for allocated IP
+    * **ip**: allocated IP for a provided domain with `InternalVIP` type in a match section
+    * **type**: provides information about the way IP was provided
+  * **addresses**: list of domains and addresses
+    * **address**: IP address, CIDR, or domain name user
+    * **status**: indicate if an address is available
+    * **reason**: holds error messages if there are any 
+
+#### Cluster name
+
+Currently, we name clusters after their respective policies. However, when dealing with multiple protocols, we find it necessary to include the port number in the cluster name.
+Example:
+
+```yaml
+kind: MeshExternalService
+metadata:
+  name: httpbin
+  namespace: kuma-system
+  labels:
+    kuma.io/mesh: default
+    kuma.io/zone: east-1
+spec:
+  match:
+  - type: Domain
+    value: httpbin.com
+    port: 80
+    protocol: http2
+  - type: Domain
+    value: httpbin.com
+    port: 443
+    protocol: tls
+ destination:
+   type: Passthrough
+```
+
+In this case, for port `80`, we require HTTP/2 configuration in the cluster settings. This necessitates having a separate configuration for each cluster.
+
+clusterName: `{policyName}_{port}`, so for the above policy name would looks like: `httpbin_80` and `httpbin_443`
+
+#### Extensability
+
+Provided model allows creating separate plugins which can integrate with different cloud providers or components. The user can create it's own custom configuration and later in the code create XDS configuration based on it. 
+
+Example:
+
+```yaml
+kind: MeshExternalService
+metadata:
+  name: example
+  namespace: kuma-system
+  labels:
+    kuma.io/mesh: default
+    kuma.io/zone: east-1
+spec:
+  match:
+  - type: InternalVIP
+    value: example.ext.svc.local
+    port: 80
+    protocol: http
+ destination:
+   type: Extension
+   extension:
+     type: Lambda 
+     config:
+       arn: arn:aws:lambda:us-west-2:123456789012:function:my-function
+```
+
+#### Universal without Transparent Proxy
+
+This specific setup has limitations caused by the nature of passthrough. 
+
+Limitations:
+* Wildcard doesn’t work through proxy
+* Only `destination.type: Manged` supported
+
+Traffic handled without transparent proxy doesn't provide information about the original destination of the request. That causes traffic to be difficult to recognize and we are not able to handle 2 different matches on the same listener, unless it's http traffic. In this case it makes sense to have only one match, or even use only destinations.
```

Comment:
At what time does this happen? When generating the configuration?

Reply:
Yes, we cannot get it earlier.